internal/gatewayapi/securitypolicy.go

```diff
@@ -151,17 +152,6 @@ func (t *Translator) ProcessSecurityPolicies(securityPolicies []*egv1a1.Security
 					continue
 				}
 
-				if err := validation.ValidateSecurityPolicy(policy); err != nil {
```

Comment:
We don't need to validate the policy again here because:
* Most of the validations are enforced by the k8s validator
* The extra validation for JWT Providers are enforced in the `validateJWTProvider` method

Reply:
this function also added some validation around api key auth that could not be represented in CEL due to the cost https://github.com/envoyproxy/gateway/blob/8b89dadfbd2cae74ab39055710688dae3c792512/api/v1alpha1/validation/securitypolicy_validate.go#L75
so I believe we cannot rm this function
cc @sanposhiho

Reply:
IMO, this should be validated through the a CEL validation.

Reply:
Specifically, I wanted to use the following CEL, but it hit this estimated cost problem:

```
((size(self.headers) > 0 ? 1 : 0) + (size(self.params) > 0 ? 1 : 0) + (size(self.cookies) > 0 ? 1 : 0)) == 1
```

which is actually very similar to the second issue above complainted.

Reply:
Thanks for the explanation! @sanposhiho 

I kept the validation for APIKeyAuth in the Gateway API translator.